add-ons/APPLICATION_TEMPLATE.md

```diff
@@ -0,0 +1,182 @@
+# New Project Project Application
+
+Please see the [proposals/new-project-addition-process.md](Documentation) to
+better understand the full process for submitting a new project.
+
+In short, copy this Application Template and populate the document.
+--!>
+
+
+## Background information
```

Comment:
Should we try to learn something from CNCF action items: https://github.com/cncf/toc/blob/main/.github/ISSUE_TEMPLATE/template-incubation-application.md#required ?
- [ ] All project metadata and resources are vendor-neutral
- [ ] Governance structure
- [ ] Contributing guides
- [ ] Overlap with existing Kubeflow projects
- [ ] Public list of adopters.